test/factory_compatibility_table_test.rb

```diff
@@ -0,0 +1,230 @@
+    false
+  rescue StandardError => e
+    debug "StandardError", description, e.full_message(highlight: true)
+    false
+  end
+
+  def basic_methods
+    @basic_methods ||= [
+      :factory,
+      :dimension,
+      :geometry_type,
+      :srid,
+      :envelope,
+      :as_text,
+      :as_binary,
+      :is_empty?,
+      :is_simple?,
+      :boundary,
+      :convex_hull,
+      :buffer
+    ].freeze
+  end
+
+  def classify_sym(sym)
+    sym.to_s.split("_").map(&:capitalize).join
+  end
+
+  def compatility_table_path
+    "#{__dir__}/../doc/Factory-Compatibility.md"
+  end
+
+  def compute_handled_methods_per_factory
+    results = {}
+    factories.each do |factory_key, factory|
+      results[factory_key] = {}
+      geometries_per_factory(factory).each do |geometry_key, geometry|
+        basic_methods.each do |method_sym|
+          description = "#{classify_sym(geometry_key)}##{method_sym}"
+          results[factory_key][description] = basic_method_handled?(geometry, method_sym, description)
+        end
+
+        relational_methods.each do |method_sym|
+          geometries_per_factory(factory).each do |other_geometry_key, other_geometry|
+            description = "#{classify_sym(geometry_key)}##{method_sym}(#{classify_sym(other_geometry_key)})"
+            results[factory_key][description] = relational_method_handled?(geometry, other_geometry, method_sym, description)
+          end
+        end
+      end
+    end
+    results
+  end
+
+  def debug(*args)
+    # puts *args
+  end
+
+  def factories
+    @factories ||= {
+      geos: RGeo::Geos.factory,
+      geos_zm: RGeo::Geos.factory(has_z_coordinate: true, has_m_coordinate: true),
+      ffi: RGeo::Geos.factory(native_interface: :ffi),
+      ffi_zm: RGeo::Geos.factory(has_z_coordinate: true, has_m_coordinate: true, native_interface: :ffi),
+      cartesian: RGeo::Cartesian.simple_factory,
+      projection: RGeo::Geographic.simple_mercator_factory,
+      spherical: RGeo::Geographic.spherical_factory
+    }.freeze
+  end
+
+  def generate_markdown(handled_per_factory_per_description)
+    descriptions = handled_per_factory_per_description.first.last.keys
+    factories = handled_per_factory_per_description.keys
+
+    sizes = [
+      descriptions.max_by(&:size).size + 2,
+      *factories.map(&:size)
+    ]
+
+    titles = ["", *factories]
+    rows = descriptions.map do |description|
+      [
+        "`#{description}`",
+        *factories.map { |factory| handled_per_factory_per_description[factory][description] ? "✅" : "❌" }
+      ]
+    end
+
+    markdown = ""
+    markdown += "| #{titles.zip(sizes).map { |title, size| title.to_s.center(size) } * ' | '} |\n"
+    markdown += "| #{ '-' * (sizes.first - 1) + ':' } | #{sizes[1..-1].map { |size| ':' + '-' * (size - 2) + ':' } * ' | '} |\n"
+    rows.each do |row|
+      markdown += "| #{row.first.rjust(sizes.first)} | #{row[1..-1].zip(sizes[1..-1]).map { |cell, size| cell.center(size) } * ' | '} |\n"
+    end
```

Comment:
This looks correct and should be working, but on my computer, it is rendering on one line. I'm on the latest version of yard and checked on both chrome and firefox and saw the same issue.

Reply:
I installed the `redcarpet` gem and didn't change anything else (just ran `gem install redcarpet`), and it rendered properly. I thought I would have to set it with the -M flag in `.yardopts`, but I guess it automatically detects one.

Do you have one installed or did it work out of the box for you?

Running `yard markups | grep markdown` shows this list of available providers.

```
[markdown]
  Providers: redcarpet rdiscount kramdown bluecloth maruku rpeg-markdown rdoc
  Extensions: .markdown .md .mdown .mkd
```

Will we need to add one of these as a dev dependency?

Reply:
@keithdoggett good catch! I do have all of these as well:

```
[markdown]
  Providers: redcarpet rdiscount kramdown bluecloth maruku rpeg-markdown rdoc
  Extensions: .markdown .md .mdown .mkd
```

We could also switch back to the html table ? (I did the markdown one for readability as a file)

I don't see documentation about markup provider in yard documentation. If you could uninstall back redcarpet and see if adding it to dev dependencies works ? If you cannot reproduce I'll do that one next week

Reply:
I couldn't make it _not work_. I'd say we publish as-is and see if rubydoc has the correct parser

Reply:
Agreed. I dug through the YARD code a little bit to see what might be causing it but I think we should just publish and if it's an issue we can deal with it then.